cmd/crank/beta/render/render.go

```diff
@@ -84,26 +85,26 @@ type Outputs struct {
 	// TODO(negz): Allow returning desired XR connection details. Maybe as a
 	// Secret? Should we honor writeConnectionSecretToRef? What if secret stores
 	// are in use?
-
-	// TODO(negz): Allow returning desired XR readiness? Or perhaps just set the
-	// ready status condition on the XR if all supplied observed resources
-	// appear ready?
 }
 
 // Render the desired XR and composed resources, sorted by resource name, given the supplied inputs.
-func Render(ctx context.Context, in Inputs) (Outputs, error) { //nolint:gocyclo // TODO(negz): Should we refactor to break this up a bit?
+func Render(ctx context.Context, logger logging.Logger, in Inputs) (Outputs, error) { //nolint:gocyclo // TODO(negz): Should we refactor to break this up a bit?
 	// Run our Functions.
 	conns := map[string]*grpc.ClientConn{}
 	for _, fn := range in.Functions {
 		runtime, err := GetRuntime(fn)
 		if err != nil {
 			return Outputs{}, errors.Wrapf(err, "cannot get runtime for Function %q", fn.GetName())
 		}
-		rctx, err := runtime.Start(ctx)
+		rctx, err := runtime.Start(ctx, logger)
```

Comment:
Would it make sense to pass the logger in `GetRuntime`? So it would be a field of the runtime struct rather than an argument to the `Start` method?